### PR TITLE
openvr: Bump deps and upgrade for conan v2

### PR DESCRIPTION
**Pull request by paulharris**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by paulharris** [comment]
Seems like a pretty cryptic error...
On Linux locally, I get this error from the test_package:
```
Unable to read VR Path Registry from /home/paul/.config/openvr/openvrpaths.vrpath
```

Which I assume is enough to test the library was able to link and execute...

File: recipes/openvr/all/conanfile.py [hunk 1]
```diff
@@ -20,83 +28,95 @@ class OpenvrConan(ConanFile):
...
 
     def package(self):
-        self.copy("LICENSE", src=os.path.join(self.source_folder, self._source_subfolder), dst="licenses")
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        self.copy(pattern="openvr_api*.dll", dst="bin", src="bin", keep_path=False)
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        copy(self, pattern="openvr_api*.dll", dst=os.path.join(self.package_folder, "bin"), src=os.path.join(self.source_folder, "bin"), keep_path=False)
```
Comment (SpaceIm) [inline recipes/openvr/all/conanfile.py:101]:
```suggestion
        copy(self, "openvr_api*.dll", src=self.build_folder, dst=os.path.join(self.package_folder, "bin"), keep_path=False)
```

File: recipes/openvr/all/conanfile.py [hunk 2]
```diff
@@ -20,83 +28,95 @@ class OpenvrConan(ConanFile):
...
 
         if self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.system_libs.append("dl")
 
-        if tools.is_apple_os(self.settings.os):
+        if is_apple_os(self):
             self.cpp_info.frameworks.append("Foundation")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        self.cpp_info.names["pkg_config"] = "openvr"
```
Comment (SpaceIm) [inline recipes/openvr/all/conanfile.py:122]:
```suggestion
```

File: recipes/openvr/all/conanfile.py [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+from conan.tools.apple import is_apple_os
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, replace_in_file, rmdir, collect_libs, get, copy
+from conan.tools.build import check_min_cppstd, stdcpp_library
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.scm import Version
 import os
-from conans import ConanFile, tools, CMake
-from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.59.0"
```
Comment (SpaceIm) [inline recipes/openvr/all/conanfile.py:10]:
```suggestion
required_conan_version = ">=1.53.0"
```

File: recipes/openvr/all/conanfile.py [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
 
 class OpenvrConan(ConanFile):
     name = "openvr"
     description = "API and runtime that allows access to VR hardware from applications have specific knowledge of the hardware they are targeting."
-    topics = ("conan", "openvr", "vr", )
+    topics = ("vr")
```
Comment (SpaceIm) [inline recipes/openvr/all/conanfile.py:15]:
```suggestion
    topics = ("vr",)
```

File: recipes/openvr/all/conandata.yml [hunk 5]
```diff
@@ -11,4 +11,17 @@ sources:
 patches:
   "1.16.8":
     - patch_file: "patches/fix-includes-and-assert-1.16.8.patch"
-      base_path: "source_subfolder"
+      patch_description: "Fix compilaton errors due to wrong headers folder and usage of missing assert.h"
+      patch_type: "backport"
```
Comment (SpaceIm) [inline recipes/openvr/all/conandata.yml:15]:
```suggestion
      patch_type: "portability"
```

File: recipes/openvr/all/conanfile.py [hunk 6]
```diff
@@ -20,83 +28,95 @@ class OpenvrConan(ConanFile):
...
+    def _min_cppstd(self):
+        return 11
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "14.1",
```
Comment (SpaceIm) [inline recipes/openvr/all/conanfile.py:39]:
```suggestion
            "msvc": "191",
```

File: recipes/openvr/all/conanfile.py [hunk 7]
```diff
@@ -20,83 +28,95 @@ class OpenvrConan(ConanFile):
...
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
```
Comment (SpaceIm) [inline recipes/openvr/all/conanfile.py:43]: Despite wrong msvc version, why do you restrict more than before? It was raisong for gcc < 5 only, so we know it works for Visual Studio 2015 for example. You should keep previous logic.

Comment (paulharris) [reply recipes/openvr/all/conanfile.py:43]: ok then, I've reverted :)

**Comment by paulharris** [comment]
I'm happy to close this in preference for #18788, which appears to supersede this PR.

**Pull request closed without merging**